hack/build-antrea-linux-all.sh

```diff
@@ -76,7 +78,10 @@ case $key in
 esac
 done
 
-THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if ! check_for_buildx; then
```

Comment:
@XinShuYang 

> Using buildx with Docker requires Docker Engine 19.03 or newer, but that version came out in 2019. Note that we already require buildx / BuildKit since merging https://github.com/antrea-io/antrea/pull/5918.

This PR does not introduce new version requirements compared to #5918. `docker buildx` has been supported since 19.03, which is almost 5 years old. Are you saying that the some builds are currently broken because of #5918, with a Docker version older than 19.03 and no possibility of upgrade?

Reply:
The docker version is 20.10.7, it showed `'buildx' is not a docker command`.

Reply:
> Are you saying that the some builds are currently broken because of #5918
#5918 caused issue #5941 due to the `--mount` compatibility problem.
This issue #5941 has been fixed by #5937, in this PR I enabled DOCKER_BUILDKIT to support `--mount` in dockerfile.

Reply:
This is probably as simple as setting the `DOCKER_CLI_EXPERIMENTAL` variable, in order to support these older versions.
Let me update the PR accordingly.

Reply:
I made that change. Could you try running `./hack/build-antrea-linux-all.sh` on one of the workers? Or maybe I can just trigger Jenkins jobs?

Reply:
> Or maybe I can just trigger Jenkins jobs?

Yes, you can trigger the cloud job manually with custom repo and branch name. e.g. https://jenkins.antrea-ci.rocks/job/cloud-antrea-gke-conformance-net-policy/build?delay=0sec 

> This is probably as simple as setting the `DOCKER_CLI_EXPERIMENTAL` variable, in order to support these older versions.

I tried to trigger one job in https://jenkins.antrea-ci.rocks/job/cloud-antrea-gke-conformance-net-policy/696/. Maybe the failure is caused by the result of `check_for_buildx`, should we run `export DOCKER_BUILDKIT=1` and `export DOCKER_CLI_EXPERIMENTAL=enabled` before running `check_for_buildx`?

Reply:
That didn't work. It's possible that the buildx plugin is not available at all if docker was installed using distribution packages and not using the Docker installation script, but I would need to ssh into the Jenkins worker to check.

Is the issue only for the public cloud Jenkins jobs (AKS / GKE / EKS)?

Reply:
This cloud node was unavailable for ssh access before the migration, but after several attempts on jenkins script, I successfully upgraded the docker. The gke tests now pass with the current code changes so I believe we can proceed with merging this PR.